pkg/proxy/ipvs/proxier.go

```diff
@@ -1692,6 +1690,23 @@ func (proxier *Proxier) syncProxyRules() {
 	proxier.deleteEndpointConnections(endpointUpdateResult.StaleEndpoints)
 }
 
+// refreshCurrentIPVSServices is responsible for updating the
+// currentIPVSServices field in the proxier object with virtual services read
+// from the ipvs interface. It should only be called when holding the proxier
+// mutex.
+func (proxier *Proxier) refreshCurrentIPVSServices() error {
+	proxier.currentIPVSServices = make(map[string]*utilipvs.VirtualServer)
+
+	currentSvcs, err := proxier.ipvs.GetVirtualServers()
+	if err != nil {
+		return err
```

Comment:
we already cleaned up the cache `proxier.currentIPVSServices` , if we fail here do we want to return wiping out the cache or keeping the old state?

Reply:
You're right, it's safer to keep it. This isn't a problem now because returning an error would cause the whole sync to be aborted, but it would be safer to only wipe `proxier.currentIPVSServices` after retrieving new data successfully to prevent problems in future changes. Fixing this on the next push.